File: modules/mqtt/src/main/kotlin/com/example/blog/mqtt/api/service/SampleMessageHandler.java
package com.example.blog.mqtt.api.service;

import org.springframework.stereotype.Component;

@Component
class SampleMessageHandler {

    fun handle(message:SampleMessage) {
        logger.info("message arrived : $message")
    }
}

data class SampleMessage(val title:String, val content:String) // (2)

File: modules/mqtt/src/main/kotlin/com/example/blog/mqtt/api/service/SampleMessageHandler.kt
package com.example.blog.mqtt.api.service

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class SampleMessageHandler { // (1)

    private val log = LoggerFactory.getLogger(SampleMessageHandler::class.java)

    fun handle(message: SampleMessage) {
        log.info("message arrived : $message")
    }
}

data class SampleMessage(val title: String, val content: String) // (2)